se/data/templates/core.css

```diff
@@ -45,6 +45,32 @@ hgroup{
 	text-align: center;
 }
 
+/* simulate h3 in an hgroup */
+hgroup h2 + p{
+	font-size: 1.17em;
+}
+
+/* simulate h4 in an hgroup */
+hgroup h2 + p + p,
+hgroup h3 + p{
+	font-size: 1em;
+}
+
+/* simulate h5 in an hgroup */
+hgroup h2 + p + p + p,
+hgroup h3 + p + p,
+hgroup h4 + p{
+	font-size: .83em;
+}
+
+/* simulate h6 in an hgroup */
+hgroup h2 + p + p + p + p,
+hgroup h3 + p + p + p,
+hgroup h4 + p + p,
+hgroup h5 + p{
+	font-size: .67em;
+}
+
```

Comment:
Also, we should add an `hgroup p` selector to line 38 so that the `<h#>` styling carries on to `<p>`

Reply:
And, we also will need `text-indent: 0` to all of these new selectors, to override `p + p` text indent styling in case we have multiple `<p>` in an `<hgroup>`

Reply:
Those h#/hgroup selectors (at 38) have a 3em margin, which would give each p in the hgroup that margin (I tested to confirm). I'm assuming we don't want that. :) Would you prefer to make a separate selector for just hgroup p and put everything but the margin on that, or add hgroup p to the existing group, remove margin, and make a separate selector for everything but hgroup p with just the margin on it?

Reply:
If we make a separate `hgroup p` selector and put everything in it but the margin, we can also add the `text-indent: 0`, so we don't have to add it to each of the other selectors. We also don't need the `text-align: center`, since that will be handled by the `hgroup` selector in the original. So:
````
hgroup p{
	break-after: avoid;
	break-inside: avoid;
	font-variant: small-caps;
	hyphens: none;
	-epub-hyphens: none;
	text-indent: 0;
}
````

Reply:
We *do* want the margins. Without a margin we won't replicate h# styling like we had before. The easiest way to replicate h# styling is to simply add `hgroup p` to the same styles that `h#` already have. We can add a separate selector to remove text indent only.

Reply:
Actually, most of those carry over to whatever's in the hgroup. The `font-variant` does; I saw that in my testing yesterday, but didn't think of it until now. The breaks should since it applies to the hgroup as a whole and therefore everything in it; I  don't know about the hyphens. So we may just need the hyphens and text-indent.

Reply:
Hyphens will inherit. It's useful to open it in a browser and use the browser's CSS inspector to see how things are being inherited and what's overriding what.

Reply:
Margins of direct siblings do _not_ collapse. This is with just `hgroup p` added to the existing selectors.
<img width="288" alt="image" src="https://github.com/standardebooks/tools/assets/19378955/fbe883c4-4270-48f0-911c-35478d864e45">

Reply:
I see, I made a mistake, though for a different reason. Line 48 removes margins for all children of `<hgroup>`. So the margins aren't important after all.

Reply:
All right, I updated the message and added the unused block, and did some more testing with the CSS.

I created a test chapter with an hgroup with h2 thru h4 elements. With the current core.css, or the new core.css, they display like this.
<img width="224" alt="image" src="https://github.com/standardebooks/tools/assets/19378955/de68a271-e15f-4991-a813-0a76eadd5939">

If I change them to p's, however, and add `hgroup p` to the selector in question, they display like this:
<img width="244" alt="image" src="https://github.com/standardebooks/tools/assets/19378955/297dc160-c75e-47d1-be89-6c958c4a578d">

Which is what I would expect. As h#s, they are caught by the `hgroup > *` selector, which zeroes their margin. We have to have that since a standalone h# _should_ have a margin, but one in an hgroup shouldn't. However, with the addition of `hgroup p` to the other selectors, that has higher precedence, and so the p's in the hgroup are caught by it rather than by `hgroup > *`, and we get margins inside the hgroup, which we don't want. Thus, we don't need the `hgroup p`; the `hgroup` handles the hgroup margins, and we don't _want_ margins within the hgroup.

There is another complication however. There is a separate set of selectors (starting with `p.continued,`) that ensure the first paragraph after an h# (and hr, etc.) aren't indented, and use hanging punctuation. Several of the selectors are of the form `h# + p`, which of course now matches the p's in the hgroup, and so they are handled by that selector. The good news is that gives us the `text-indent: 0` we wanted; the other news is that that also gives them hanging punctuation, which maybe we want in the header and maybe we don't, and maybe it doesn't matter since they're centered?

Reply:
Sorry, I didn't see your followup message until just now. So the only new news is the above complication.